### PR TITLE
Mlcube download logging

### PR DESCRIPTION
**Pull request by VukW**
Catches `docker pull` output and put it directly to medperf's stdout:
image
The result is a bit weird, but at least it shows the `docker pull` is running and smth is executing

### CODE REVIEW

File: cli/medperf/entities/cube.py [hunk 1]
```diff
@@ -220,13 +221,16 @@ def download_image(self):
             # Retrieve image from image registry
             logging.debug(f"Retrieving {self.id} image")
             cmd = f"mlcube configure --mlcube={self.cube_path}"
-            with pexpect.spawn(cmd, timeout=config.mlcube_configure_timeout) as proc:
-                proc_out = proc.read()
+            cmd_env = {**os.environ, 'DOCKER_CLI_HINTS': 'false'}
+            with pexpect.spawn(cmd, timeout=config.mlcube_configure_timeout, env=cmd_env) as proc:
+                proc.logfile_read = DebugAndOutputLogger()  # Redirect cmd output to medperf stdout
```
Comment (VukW) [inline cli/medperf/entities/cube.py:226]: I was exploring to use `combine_proc_sp_text` there to display docker logs in a handy way. However, this function does not work with non-linear output (i.e. when output is modifying in the process with escape sequences), and I wasn't successful to fix that. Thus, offering the current, a few ugly, but still solution

File: cli/medperf/entities/cube.py [hunk 2]
```diff
@@ -220,13 +221,16 @@ def download_image(self):
             # Retrieve image from image registry
             logging.debug(f"Retrieving {self.id} image")
             cmd = f"mlcube configure --mlcube={self.cube_path}"
-            with pexpect.spawn(cmd, timeout=config.mlcube_configure_timeout) as proc:
-                proc_out = proc.read()
+            cmd_env = {**os.environ, 'DOCKER_CLI_HINTS': 'false'}
```
Comment (VukW) [inline cli/medperf/entities/cube.py:224]: this env hides `What's next?` docker CLI hits from its output, leaving only essential logs about pulling docker layers

**Comment by VukW** [comment]
Prettified output according to guidelines
image

File: cli/medperf/utils.py [hunk 3]
```diff
@@ -245,6 +246,27 @@ def combine_proc_sp_text(proc: spawn) -> str:
...
+    def __init__(self):
+        self._just_started_flag = True
+
+    def write(self, byte):
+        if self._just_started_flag:
+            sys.stdout.write("\r\033[K")  # clean the current spinner line
+            self._just_started_flag = False
+        s = byte.decode("utf-8")
+        sys.stdout.write(f'{Fore.WHITE}{Style.DIM}{s}{Style.RESET_ALL}')
+        sys.stdout.flush()
```
Comment (VukW) [inline cli/medperf/utils.py:262]: The `yasmin` spinner interferes too much into the printed lines to ensure statis text is always shown. That's why it's impossible to print dynamically changed logs with escape sequences via `config.ui.write()` and I had to print out everything manually via `sys.stdout`

**Comment by hasan7n** [comment]
@VukW should we close this?

**Pull request closed without merging**